ovn-tester/cms/openstack/tests/base_openstack.py

```diff
@@ -36,4 +37,4 @@ def run(self, ovn: OpenStackCloud, global_cfg):
                 worker_node.provision(ovn)
 
             for _ in range(self.config.n_projects):
-                _ = ovn.new_project(gw_nodes=ovn.worker_nodes)
+                _ = ovn.new_project(gw_nodes=self.config.n_gws_per_project)
```

Comment:
I wonder if we should find a more specific name for this configuration option. While it fits well with the current implementation, it does not really fit with how it could be used in OVN.

The `Gateway_Chassis` (or possibly `HA_Chassis_Group` at some point in the future) is a property of the [`Logical_Router_Port`](https://manpages.ubuntu.com/manpages/jammy/man5/ovn-nb.5.html#logical_router_port%20table).

A project can have multiple routers, and routers can have multiple logical router ports with different configuration for these attributes. There is [ongoing work in OpenStack to support multiple LRPs per router](https://specs.openstack.org/openstack/neutron-specs/specs/2023.2/active-active-l3-gateway-with-multihoming.html).

Would perhaps `n_chassis_per_gw_lrp` work?

Reply:
hmm. If we call it `n_chassis_per_gw_rlp`, do we also need to tie it back to the projects with something like `n_gw_ports_per_project` ?

Reply:
> hmm. If we call it `n_chassis_per_gw_rlp`, do we also need to tie it back to the projects with something like `n_gw_ports_per_project` ?

I'm very picky on configuration option naming, as opposed to implementations which change frequently, options are a kind of API and have a infinite lifetime.  Any future changes to their name or meaning may have expensive workflow change costs down the line.

So my intention with this comment was scoped at this specific option.

But you are right in that it would probably also make sense to have other options to control the adjacent entities in the project-router-lrp test matrix.

Would something like `n_gw_ports_per_router` and `n_routers_per_project` catch more use cases?

Not sure if we need them right now though.

Reply:
You are right. We need to strike balance between future-proofing and not inventing unnecessary options here. We can stick with `n_chassis_per_gw_lrp` with implication that currently each project has only one router and one gw port.